### PR TITLE
MGMT-10849: Analyze must-gather using Insights Core

### PR DESCRIPTION
**Pull request by mkowalski**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -21,3 +21,6 @@ sh==1.14.2
 semver==2.13.0
 pytest==7.1.1
 networkx==2.5.1
+insights-core>3.0.276
+git+https://gitlab.cee.redhat.com/ccx/ccx-ocp-core.git
```
Comment (osherdp) [inline requirements.txt:25]: Does it work without VPN access?

Comment (mkowalski) [reply requirements.txt:25]: No, it's a very private repository that must not be available from outside

Comment (osherdp) [reply requirements.txt:25]: So I think it's not fitting into this repo...
Unless somehow we make its installation optional in some way

Comment (vrutkovs) [reply requirements.txt:25]: Agree, it needs to be optional

Comment (mkowalski) [reply requirements.txt]: Ok, extracted to a separate requirements file that will be handled by https://github.com/openshift/assisted-test-infra/pull/1712 inside our Jenkins

File: Dockerfile.assisted-installer-deployment [hunk 2]
```diff
@@ -28,7 +28,9 @@ COPY dev-requirements.txt /tmp/dev-requirements.txt
 RUN pip install -r /tmp/dev-requirements.txt
 
 COPY requirements-rht.txt /tmp/requirements-rht.txt
+RUN git config --global http.sslVerify false
 RUN pip install -r /tmp/requirements-rht.txt || true
+RUN git config --global http.sslVerify true
```
Comment (mkowalski) [inline Dockerfile.assisted-installer-deployment:33]: This is a hack (or a trick) to bypass the problem that underlying `pip install` runs `git clone` over HTTPS but we do not have this custom CA cert in the system store. As pips trusted-host parameter does not work when there is a nested git running, I did not find a smarter way to do what is being done here.

File: tools/add_triage_signature.py [hunk 3]
```diff
@@ -1061,6 +1080,46 @@ def _update_ticket(self, url, issue_key, should_update=False):
...
+        md = get_metadata_json(url)
+        cluster_id = md['cluster']['id']
+        triage_logs_tar = get_triage_logs_tar(triage_url=url, cluster_id=cluster_id)
+
+        try:
+            mustgather = get_mustgather(triage_logs_tar)
+        except FailedToGetMustgatherException:
+            return
+
+        tmp_mustgather = tempfile.NamedTemporaryFile(suffix=".tar.gz", delete=False)
```
Comment (omertuc) [inline tools/add_triage_signature.py:1102]: This should also be `with tempfile.Named...`

File: tools/add_triage_signature.py [hunk 4]
```diff
@@ -1061,6 +1080,43 @@ def _update_ticket(self, url, issue_key, should_update=False):
...
+        logger.debug(f"Writing must-gather as {tmp_mustgather.name}")
+        try:
+            tmp_mustgather.write(mustgather)
+            report = subprocess.run(["insights", "run", "-p", "ccx_rules_ocp", tmp_mustgather.name], stdout=subprocess.PIPE)
+            with tempfile.NamedTemporaryFile(prefix="insights-report-", suffix=".txt") as tmp_report:
+                try:
```
Comment (omertuc) [inline tools/add_triage_signature.py:1108]: lol you applied my suggestion before I edited it :man_facepalming:

Comment (omertuc) [reply tools/add_triage_signature.py:1108]: See my most recent edit for what I truly meant...

File: Dockerfile.assisted-installer-deployment [hunk 5]
```diff
@@ -28,6 +28,12 @@ COPY dev-requirements.txt /tmp/dev-requirements.txt
 RUN pip install -r /tmp/dev-requirements.txt
 
 COPY requirements-rht.txt /tmp/requirements-rht.txt
+# required in order to handle self-signed repository
+RUN curl -k -d op=download -d mimeType=application/x-x509-ca-cer \
+    -d submit=Submit https://ca01.pki.prod.int.rdu2.redhat.com:8443/ca/ee/ca/getCAChain \
+    | openssl pkcs7 -inform DER -print_certs \
+    -out /etc/pki/ca-trust/source/anchors/Red_Hat_CA_chain.pem || true
+RUN update-ca-trust || true
 RUN pip install -r /tmp/requirements-rht.txt || true
```
Comment (mkowalski) [inline Dockerfile.assisted-installer-deployment:37]: Seems to have worked
```
17:36:38  Step 13/20 : RUN curl -k -d op=download -d mimeType=application/x-x509-ca-cer     -d submit=Submit https://ca01.pki.prod.int.rdu2.redhat.com:8443/ca/ee/ca/getCAChain     | openssl pkcs7 -inform DER -print_certs     -out /etc/pki/ca-trust/source/anchors/Red_Hat_CA_chain.pem || true

17:36:38   ---> Running in 52e4a28ca828
17:36:39    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
17:36:39                                   Dload  Upload   Total   Spent    Left  Speed

17:36:39  
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  3145  100  3085  100    60   3364     65 --:--:-- --:--:-- --:--:--  3425
17:36:40  Removing intermediate container 52e4a28ca828
17:36:40   ---> 38702ce2462d
17:36:40  Step 14/20 : RUN update-ca-trust || true
17:36:40   ---> Running in 78e7ccd3241f

17:36:41  Removing intermediate container 78e7ccd3241f
17:36:41   ---> cb1a40f2f888
17:36:41  Step 15/20 : RUN pip install -r /tmp/requirements-rht.txt || true
17:36:41   ---> Running in 4b92466bfdc0

17:36:42  Collecting git+https://gitlab.cee.redhat.com/ccx/ccx-ocp-core.git (from -r /tmp/requirements-rht.txt (line 1))
17:36:42    Cloning https://gitlab.cee.redhat.com/ccx/ccx-ocp-core.git to /tmp/pip-req-build-vmt3vowt
17:36:42    Running command git clone --filter=blob:none --quiet https://gitlab.cee.redhat.com/ccx/ccx-ocp-core.git /tmp/pip-req-build-vmt3vowt

17:36:46    Resolved https://gitlab.cee.redhat.com/ccx/ccx-ocp-core.git to commit 478764d6807d2a95dcafeed9250ef0e6280fed82
17:36:46    Preparing metadata (setup.py): started

17:36:48    Preparing metadata (setup.py): finished with status 'done'
17:36:48  Collecting git+https://gitlab.cee.redhat.com/ccx/ccx-rules-ocp.git (from -r /tmp/requirements-rht.txt (line 2))
17:36:48    Cloning https://gitlab.cee.redhat.com/ccx/ccx-rules-ocp.git to /tmp/pip-req-build-q2cwymzc
17:36:48    Running command git clone --filter=blob:none --quiet https://gitlab.cee.redhat.com/ccx/ccx-rules-ocp.git /tmp/pip-req-build-q2cwymzc

17:37:20    Resolved https://gitlab.cee.redhat.com/ccx/ccx-rules-ocp.git to commit 459831a94d4cc200023ef013f22e8ca944659cd8
17:37:20    Preparing metadata (setup.py): started
17:37:20    Preparing metadata (setup.py): finished with status 'done'
```

File: tools/add_triage_signature.py [hunk 6]
```diff
@@ -1061,6 +1080,46 @@ def _update_ticket(self, url, issue_key, should_update=False):
...
+        md = get_metadata_json(url)
+        cluster_id = md['cluster']['id']
+        triage_logs_tar = get_triage_logs_tar(triage_url=url, cluster_id=cluster_id)
+
+        try:
+            mustgather = get_mustgather(triage_logs_tar)
+        except FailedToGetMustgatherException:
+            return
+
+        with tempfile.NamedTemporaryFile(suffix=".tar.gz", delete=False) as tmp_mustgather:
```
Comment (omertuc) [inline tools/add_triage_signature.py:1102]:
```suggestion
        with tempfile.NamedTemporaryFile(suffix=".tar.gz") as tmp_mustgather:
```

File: tools/add_triage_signature.py [hunk 7]
```diff
@@ -1061,6 +1080,46 @@ def _update_ticket(self, url, issue_key, should_update=False):
...
+                        logger.debug(f"Writing Insights report as {tmp_report.name}")
+                        tmp_report.write(report.stdout)
+                        tmp_report.flush()
+                        self._upload_attachment(issue_key, tmp_report.name)
+                    except Exception:
+                        logger.exception(f"Error while writing and uploading {tmp_report.name}")
+            except Exception:
+                logger.exception(f"Error while handling must-gather in {tmp_mustgather.name}")
+
+        report = "*Insights report:* {}\n".format(tmp_report.name)
```
Comment (omertuc) [inline tools/add_triage_signature.py:1119]: "See attachment" ?

File: tools/add_triage_signature.py [hunk 8]
```diff
@@ -1061,6 +1080,46 @@ def _update_ticket(self, url, issue_key, should_update=False):
...
+                tmp_mustgather.flush()
+                report = subprocess.run(["insights", "run", "-p", "ccx_rules_ocp", tmp_mustgather.name], stdout=subprocess.PIPE)
+                with tempfile.NamedTemporaryFile(prefix="insights-report-", suffix=".txt") as tmp_report:
+                    try:
+                        logger.debug(f"Writing Insights report as {tmp_report.name}")
+                        tmp_report.write(report.stdout)
+                        tmp_report.flush()
+                        self._upload_attachment(issue_key, tmp_report.name)
+                    except Exception:
+                        logger.error(f"Error while writing and uploading {tmp_report.name}")
```
[the remark this answers is not in the document]
Comment (mkowalski) [reply tools/add_triage_signature.py]: Ok

File: tools/add_triage_signature.py (hunk 7 above)
Comment (mkowalski) [reply tools/add_triage_signature.py]: Ok

**Pull request merged**